kitchen-tests/cookbooks/end_to_end/recipes/_chef_client_trusted_certificate.rb

```diff
@@ -1,10 +1,10 @@
 chef_client_trusted_certificate "self-signed.badssl.com" do
```

Comment:
OK this is painful, lets stop doing this every few months.

Instead let's do something like:

```ruby
# First grab the cert. While this wouldn't ordinarily be secure, this isn't
# trying to secure something, we simply want to make sure that if we
# have said a certificate is trusted, it will be trusted. So lets grab it, trust
# it, and then try to use it.

# First, grab it
out = Mixlib::ShellOut.new(
  %w{openssl s_client -showcerts -connect self-signed.badssl.com:443}
).run_command.stdout

cert = Mixlib::ShellOut.new(%w{openssl x509 -notext}, :input => out).run_command.stdout

# Second trust it
chef_client_trusted_certificate "self-signed.badssl.com" do
  certificate cert
end

# Now try to access that site
remote_file ::File.join(Chef::Config[:file_cache_path], "index.html") do
  source "https://self-signed.badssl.com/index.html"
end
```

Reply:
done